tests/checkers/unittest_unicode.py

```diff
@@ -0,0 +1,904 @@
+        """
+        The static function _get_modules_names shall resolve module names correctly
+        """
+        assert self.CHECKER_CLASS._get_module_names("foo.bar.test") == [
+            "foo",
+            "bar",
+            "test",
+        ]
+        assert self.CHECKER_CLASS._get_module_names("foo.bar", "test") == [
+            "foo",
+            "bar",
+            "test",
+        ]
+
+    @pytest.mark.skipif(
+        sys.version_info < (3, 8), reason="requires python3.8 or higher"
+    )
+    def test_kwargs_and_position_only(self):
+        """
+        Even the new position only and keyword only should be found
+        """
+        node = astroid.extract_node(
+            """
+                def name(
+                    ok,
+                    not_økay,
+                    not_okay_defaułt=None,
+                    /,
+                    p_or_kw_okay=None,
+                    p_or_kw_not_økay=None,
+                    *,
+                    kw_arg_okay,
+                    kw_arg_not_økay,
+                ):
+                    ...
+            """
+        )
+        assert isinstance(node, nodes.FunctionDef)
+        arguments: nodes.Arguments = node.args
+
+        posargs = list(arguments.posonlyargs)
+        args = list(arguments.args)
+        kwargs = list(arguments.kwonlyargs)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id=self.EXPECTED_MSG,
+                node=posargs[1],
+                args=("Argument", "not_økay"),
+                confidence=pylint.interfaces.HIGH,
+                line=4,
+                col_offset=4,
+                end_line=4,
+                end_col_offset=13,
+            ),
+            pylint.testutils.MessageTest(
+                msg_id=self.EXPECTED_MSG,
+                node=posargs[2],
+                args=("Argument", "not_okay_defaułt"),
+                confidence=pylint.interfaces.HIGH,
+                line=5,
+                col_offset=4,
+                end_line=5,
+                end_col_offset=21,
+            ),
+            pylint.testutils.MessageTest(
+                msg_id=self.EXPECTED_MSG,
+                node=args[1],
+                args=("Argument", "p_or_kw_not_økay"),
+                confidence=pylint.interfaces.HIGH,
+                line=8,
+                col_offset=4,
+                end_line=8,
+                end_col_offset=21,
+            ),
+            pylint.testutils.MessageTest(
+                msg_id=self.EXPECTED_MSG,
+                node=kwargs[1],
+                args=("Argument", "kw_arg_not_økay"),
+                confidence=pylint.interfaces.HIGH,
+                line=11,
+                col_offset=4,
+                end_line=11,
+                end_col_offset=20,
+            ),
+        ):
+            self.checker.visit_functiondef(node)
+
+    @pytest.mark.parametrize(
+        "code, assign_type, line, col_start, col_end",
+        [
```

Comment:
This would be a lot more maintainable as functional tests, what do you think @DanielNoord ?

Reply:
There are already functional test for this test cases.
But this is a unit test, where I want to assure that i.e. the correct node is found.

Reply:
No the tests are fine, but the code content could be moved to functional maybe.

```python
                def name(
                    ok,
                    not_økay,  # [non-ascii-name]
                    not_okay_defaułt=None,  # [non-ascii-name]
                    /,
                    p_or_kw_okay=None,
                    p_or_kw_not_økay=None,  # [non-ascii-name]
                    *,
                    kw_arg_okay,
                    kw_arg_not_økay,  # [non-ascii-name]
                ):
                    ...
```      
In a functional test file is equivalent to the unit test above and easier to maintain, unless I'm missing something that is tested in the unit test..

Reply:
I think what @CarliJoy is referring to is that the functional test never actually test if we got the correct `node`. We only make this assumption because if the line and column numbering is correct, the node will probably have been correct (especially if the `add_message` does not provide them itself.

Since this message can't pass a `node` the line numbering is always set by us. Therefore, the assumption made in the functional tests is "less true". Personally, I would be okay with moving this to functional since the `add_message` still uses `node.line` probably so the assumption is still mostly true.

Or at least, that was how I interpreted @CarliJoy's message. I could be totally off 😅

Reply:
The quoted code is already here: `tests/functional/u/unicode/unicode_homoglyph_pos_and_kwonly_function.py`

@DanielNoord got me correctly.

I did some small change to allow to check for the correct node but skip all the position arguments, that will be determined by some other code parts mostly (if not set manually).

Reply:
Ok, let's keep it that way then.